cmd/api/src/api/auth.go

```diff
@@ -329,6 +330,56 @@ func SetSecureBrowserCookie(request *http.Request, response http.ResponseWriter,
 	})
 }
 
+func (s authenticator) CreateSSOSession(request *http.Request, response http.ResponseWriter, principalNameOrEmail string, authProvider any) {
+	var (
+		hostURL    = *ctx.FromRequest(request).Host
+		requestCtx = request.Context()
+	)
+
+	if user, err := s.db.LookupUser(requestCtx, principalNameOrEmail); err != nil {
+		if !errors.Is(err, database.ErrNotFound) {
+			HandleDatabaseError(request, response, err)
+		} else {
+			WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusForbidden, "user is not allowed", request), response)
+		}
+	} else {
+		switch typedAuthProvider := authProvider.(type) {
+		case model.SAMLProvider:
+			if !user.SAMLProviderID.Valid || typedAuthProvider.ID != user.SAMLProvider.ID {
+				WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusForbidden, "user is not allowed", request), response)
+				return
+			}
+		case model.OIDCProvider:
+			//todo connect to db provider table
+			break
```

Comment:
This must be done once user table is properly migrated / updated to handle either a generic sso_provider_id or a second oidc_provider_id table @mvlipka

Reply:
Will need to be updated with the patch user PR